src/components/SignIn/SignIn.jsx

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import css from "./SignIn.module.css";
+import {customHistory} from "../../App";
+
+export class SignIn extends React.Component{
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            form: {
+                email: '',
+                password: ''
+            },
+            responseStatus: ''
+        }
+    }
+
+    onChange = (event) => {
+      this.setState({
+          form: {
+              ...this.state.form,
+              [event.target.name]: event.target.value
+          }
+      })
+    };
+
+    onSubmit = (event) => {
+        if(event) {
+            event.preventDefault();
+        }
+
+        fetch('/auth/login', {
+            method: 'POST',
+            headers: {
+                'content-type': 'application/json'
+            },
+            body: JSON.stringify(this.state.form)
+        })
+            .then(response => {
```

Comment:
The problem that in general, you have no guarantees that `setState` will be executed before the next `then`, so it might be that you will check status before it is actually set...

Alternative variant.
```
.then(response => {
        if (response.status !== 200) {
            throw ...
        }
                return response.json();
            })
            .then(response => {
                    localStorage.setItem('TOKEN', response.accessToken);
                    this.props.onChangeFlag(true);
                    customHistory.push('/dashboard');
            })
```